src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,174 @@
+    private static final String RECIPE_MISSING_TAG = "Recipe is missing \"TAG\"!\n";
+    private static final String RECIPE_MISSING_STEP = "Recipe is missing \"SUM of the STEPs\"!\n";
+    /**
+     * Returns a <code>Command</code> type which contains the command to be
+     * executed and its full description to support its execution. It takes
+     * in a string representing the user input.
+     *
+     * @param line the input line by user.
+     * @return a <code>Command</code> containing the command to be executed.
+     */
+    public static Command parseCommands(String line) {
+        String[] lineSpaced = line.split(" ");
+        String fullDescription = "";
+        if (lineSpaced.length > 1) {
+            for (int i = 2; i < lineSpaced.length; i++) {
+                lineSpaced[1] = lineSpaced[1].concat(" " + lineSpaced[i]);
+            }
+            fullDescription = lineSpaced[1].trim();
+        }
+        CommandType type;
+        switch (lineSpaced[0]) {
+        case "list":
+            type = CommandType.LIST;
+            break;
+        case "add":
+            type = CommandType.ADD;
+            break;
+        case "view":
+            type = CommandType.VIEW;
+            break;
+        case "find":
+            type = CommandType.FIND;
+            break;
+        case "editstep":
+            type = CommandType.EDITSTEP;
+            break;
+        case "editingredient":
+            type = CommandType.EDITINGREDIENT;
+            break;
+        case "delete":
+            type = CommandType.DELETE;
+            break;
+        case "help":
+            type = CommandType.HELP;
+            break;
+        case "clear":
+            type = CommandType.CLEAR;
+            break;
+        case "exit":
+            type = CommandType.EXIT;
+            break;
+        default:
+            type = CommandType.UNKNOWN;
+        }
+        return new Command(type, fullDescription);
+    }
+
+
+    private static boolean matchString(String input, String regex) {
+        String matcher = input;
+        int count = 0;
+        while (matcher.contains(regex)){
+            matcher=matcher.substring(matcher.indexOf(regex)+1);
+            ++count;
+        }
+        boolean isMatch = (count==1);
+        return isMatch;
+    }
+    /**
+     * Returns an Array of Strings containing the parsed full description
+     * of a <code>Recipe</code> into its name, ingredients and tag.
+     *
+     * @param description the full description of the <code>Recipe</code>.
+     * @return parsed ArrayList of Strings containing the recipe's name, ingredients list and tag.
+     * @throws IncompleteInputException if full description input is missing the description or due date or both.
+     */
+    public static ArrayList<String> parseRecipe(String description) throws IncompleteInputException {
+        ArrayList<String> parsed = new ArrayList<>();
```

Comment:
You could follow more conventional naming for this variable. To my understanding these are your parsed command parameters, perhaps something a little more descriptive? :)